src/Frame.js

```diff
@@ -0,0 +1,35 @@
+class Frame {
+
+  constructor() {
+    this.pins = []
+  }
+
+  add(pins) {
+    this.pins.push(pins);
+  }
+
+  _isComplete() {
+    return this.pins.length === 2 || this._isaStrike()
+  }
+
+  result() {
+    if (this._isaStrike()) {
+      return 10
+    } else {
+      return this.pins[0] + this.pins[1]
+    }
+  }
+
+  _isaStrike() {
```

Comment:
Nice use of private methods!

From [this resource on OOP](http://roboprog010.github.io/posts/technical/week5.html) :
>Encapsulation just means that all of the implementation details of the object being modeled are hidden and only the methods and properties required to use the object are exposed

By making any methods that aren't required to use the object private, you are increasing encapsulation and reducing the risk that another object will mess up the state of this object

Reply:
I'm not sure what you mean here. Do you mean only methods that need to be called directly by the user should be public (I think in this challenge that would be only game.roll(n)) or should this also apply to methods that the interface needs to call on (eg game.isOver() or game.framecount() which returns the number of the frame currently being played)?

Reply:
Having had a look at the OOP I think it should just be game.roll...

Reply:
Private methods should not be called from outside of a class. Does this guide you to the answer to the question?

Reply:
So private methods should not be called from anything outside of the model? (I'm pretty sure you don't mean they should not be called from outside the class in which they themselves are defined...)
So this has led me to another question, is the interface.js file considered to be a part of the model, or is it the controller? I think it's the latter?

Reply:
[This page](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Classes/Private_class_fields) is some documentation about private class fields in JS. Just like in Ruby, you shouldn't call private class methods from outside of the class. 


For example, in the Thermostat example, we don't call the private method [`_isMinimumTemperature`](https://github.com/makersacademy/thermostat-example/blob/bfd88a9e50687aa462b7fabb0c777ee6877bafcc/src/Thermostat.js#L54) from the [thermostat-example interface](https://github.com/makersacademy/thermostat-example/blob/master/src/interface.js), we only call the public methods in the Thermostat class